tests/foreman/api/test_usergroup.py

```diff
@@ -0,0 +1,275 @@
+            entities.UserGroup().create() for _ in range(randint(3, 5))]
+        user_group = entities.UserGroup(usergroup=sub_user_groups).create()
+        self.assertEqual(
+            set(usergroup.name for usergroup in sub_user_groups),
+            set(usergroup.read().name for usergroup in user_group.usergroup)
+        )
+
+    @tier1
+    def test_negative_create_with_name(self):
+        """Attempt to create user group with invalid name.
+
+        @Feature: Usergroup
+
+        @Assert: User group is not created.
+        """
+        for name in invalid_values_list():
+            with self.subTest(name):
+                with self.assertRaises(HTTPError):
+                    entities.UserGroup(name=name).create()
+
+    @tier1
+    def test_negative_create_with_same_name(self):
+        """Attempt to create user group with a name of already existent entity.
+
+        @Feature: Usergroup
+
+        @Assert: User group is not created.
+        """
+        user_group = entities.UserGroup().create()
+        with self.assertRaises(HTTPError):
+            entities.UserGroup(name=user_group.name).create()
+
+    @tier1
+    def test_positive_update(self):
+        """Update existing user group with different valid names.
+
+        @Feature: Usergroup
+
+        @Assert: User group is updated successfully.
+        """
+        user_group = entities.UserGroup().create()
+        for new_name in valid_data_list():
+            with self.subTest(new_name):
+                user_group.name = new_name
+                user_group = user_group.update(['name'])
+                self.assertEqual(new_name, user_group.name)
+
+    @tier1
+    def test_positive_update_with_new_user(self):
+        """Add new user to user group
+
+        @Feature: Usergroup
+
+        @Assert: User is added to user group successfully.
+        """
+        user = entities.User().create()
+        user_group = entities.UserGroup().create()
+        user_group.user.append(user)
+        user_group = user_group.update(['user'])
+        self.assertEqual(user.login, user_group.user[0].read().login)
+
+    @tier2
+    def test_positive_update_with_existing_user(self):
+        """Update user that assigned to user group with another one
+
+        @Feature: Usergroup
+
+        @Assert: User group is updated successfully.
+        """
+        users = [entities.User().create() for _ in range(2)]
+        user_group = entities.UserGroup(user=[users[0]]).create()
+        user_group.user[0] = users[1]
+        user_group = user_group.update(['user'])
+        self.assertEqual(users[1].login, user_group.user[0].read().login)
+
+    @tier1
+    def test_positive_update_with_new_role(self):
+        """Add new role to user group
+
+        @Feature: Usergroup
+
+        @Assert: Role is added to user group successfully.
+        """
+        new_role = entities.Role().create()
+        user_group = entities.UserGroup().create()
+        user_group.role.append(new_role)
```

Comment:
I'm curious why you started using `append` instead of simple `user_group.role = [new_role]`?
Think of what will happen, for example, if usergroup will have 1 pre-created role? With ``user_group.role = [new_role]` you could just assert the first element of the list (like you did), but with `append` as you're adding the element to the end of list you should either use `assertIn` or check the latest element (`user_group.role[-1]`). Imo better to keep using our common `user_group.role = [new_role]`.